File: 0063-unique-paths-ii/0063-unique-paths-ii.py
class Solution(object):
    def uniquePathsWithObstacles(self, obstacleGrid):
        """
        :type obstacleGrid: List[List[int]]
        :rtype: int
        """

        if obstacleGrid[0][0] == 1:
            return 0
        obstacleGrid[0][0] = 1
        
        for n in range(1, len(obstacleGrid[0])):
            if obstacleGrid[0][n] == 0:
                obstacleGrid[0][n] += obstacleGrid[0][n-1]
            else:
                obstacleGrid[0][n] = 0
        for n in range(1,len(obstacleGrid)):
            if obstacleGrid[n][0] == 0:
                obstacleGrid[n][0] += obstacleGrid[n-1][0]
            else:
                obstacleGrid[n][0] = 0
        for n in range(1, len(obstacleGrid)):
            for m in range(1, len(obstacleGrid[n])):
                if obstacleGrid[n][m] == 0:
                    obstacleGrid[n][m] = obstacleGrid[n-1][m] + obstacleGrid[n][m-1]
                else:
                    obstacleGrid[n][m] = 0
        return obstacleGrid[-1][-1]

File: 0063-unique-paths-ii/0063-unique-paths-ii.java
class Solution {
    public int uniquePathsWithObstacles(int[][] obstacleGrid) {
        int[] checker = new int[obstacleGrid[0].length];
        Boolean found = false;
        for(int i = 0; i < obstacleGrid[0].length; i++){
            if(obstacleGrid[0][i] == 1 && !found){
                found = true;
                checker[i] = 0;
            }
            else if(!found){
                checker[i] = 1;
            }
            else{
                checker[i] = 0;
            }
        }
        
        for(int i = 1; i< obstacleGrid.length; i++){
            if(obstacleGrid[i][0] == 1){
                checker[0] = 0;
            }
            else{
                ;
            }
            for(int x = 1; x < obstacleGrid[i].length; x++){
                if(obstacleGrid[i][x] == 1){
                    checker[x] = 0;
                }
                else{
                    checker[x] += checker[x-1];
                }
            }
        }
        return checker[checker.length - 1];
        
    }
}